### DISCUSSION TITLE
I want to publish a observablehq web page to the public web. I already purchased a domain name. What are the steps?

### DISCUSSION THREAD

**Discussion by Daircom** [observablehq/framework#1482] [Q&A] [answered]
So I have my webapp via observablehq. I bought a domain name for the public web. And I want to publish my observablehq web app to the public web. But I do not have experience with creating public websites. What are the steps I have to take? I am open to use any tool.

Kind regards,
Bas

**Comment by mythmon** [comment, marked as answer]
The easiest thing you can do to make your project visible publicly is with `npm run deploy`, which will deploy the project to a URL on the observablehq.cloud domain. If you choose to make the project public then anyone will be able to use the URL to see the project. You can learn more about this option here: https://observablehq.com/framework/deploying. For example, I have a public at https://mythmon.observablehq.cloud/workbench/. (There isn't anything too exciting there yet though)

The above will not be able to use the domain name that you purchased however. Custom domains are not currently a feature that Observable supports. You'd have to use a third party web host, such as GitHub pages or Vercel. We are considering this as a future feature though, so stay tuned. You can see some very brief tips about how to go about doing this at the end of the [deploy docs](https://observablehq.com/framework/deploying#other-hosting-services) mentioned above.

**Reply by Daircom** [reply]
Hi @mythmon,

Thanks a lot! Great answer! Enough for me to get started!

**Reply by jaanaltosaar** [reply]
I'm happy to help as well @Daircom ! I made my website (jaan.li) with observable framework, and we have several demos here: 

https://onefact.github.io/new-york-real-estate/

https://onefact.github.io/healthcare-data/

https://onefact.github.io/american-community-survey/new-york-area/income-by-race

Hope that helps! Ping me if you have any issues deploying to GitHub Pages (we have also tested Cloudflare Pages, but that is less reliable).
